app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -82,7 +83,12 @@ public void newSurvey(Activity activity) {
 
     @Override
     public void sendSurvey() {
-        Session.getMalariaSurveyDB().updateSurveyStatus();
+        SurveyDB malariaSurvey = Session.getMalariaSurveyDB();
+        malariaSurvey.updateSurveyStatus();
+        if (malariaSurvey.isCompleted()) {
+            malariaSurvey.setEventUid(String.valueOf(UIDGenerator.generateUID()));
```

Comment:
We have to change a lot of files to encapsule it into  a domain entity, because Session returns DB surveys.